python/taulabs/telemetry.py

```diff
@@ -327,13 +327,20 @@ def _do_io(self, finish_time):
         if r:
             # Shouldn't throw an exception-- they just told us
             # it was ready for read.
-            chunk = os.read(self.fd, 1024)
-            if chunk == '':
-                raise RuntimeError("stream closed")
+            # TODO: Figure out why read sometimes fails when using sockets
+            try:
+                chunk = os.read(self.fd, 1024)
+                if chunk == '':
+                    raise RuntimeError("stream closed")
 
-            self.recv_buf = self.recv_buf + chunk
+                self.recv_buf = self.recv_buf + chunk
```

Comment:
Yah, realized it's correct. But it's annoyingly hard to disambiguate between accidental IDE style changes and intentional scope changes. No matter, just ranting.